common4j/src/main/com/microsoft/identity/common/java/platform/AbstractDevicePopManager.java

```diff
@@ -986,6 +992,22 @@ private String mintSignedHttpRequestInternal(@Nullable final String httpMethod,
             exception = e;
             errCode = KEYSTORE_NOT_INITIALIZED;
         } catch (final JOSEException e) {
+            if ((e.getMessage() != null && e.getMessage().contains(NEGATIVE_THIRTY_THREE_INTERNAL_ERROR)) ||
+                    (e.getCause() != null && e.getCause().getMessage().contains(NEGATIVE_THIRTY_THREE_INTERNAL_ERROR)) ||
```

Comment:
can we move all these checks to isNegativeInternalErrorInvalidKeyException? pass exception directly to the method

Reply:
+1

Reply:
Right now the code is too confusing with similar checks happening in multiple places..so as suggested let's move it all in one method

Reply:
ack